### PR TITLE
DOCS-2653: Fix links in Modin's documentation

### LINKED ISSUE #2653: Fix working of links in Modin's documentation
### Describe the problem

Several links in https://modin.readthedocs.io/en/latest/ don't work.

### PR DESCRIPTION
**Pull request by prutskov**
Signed-off-by: Alexey Prutskov <alexey.prutskov@intel.com>

## What do these changes do?

This changes add new way of linking internal pages and fix some not working links.
- [ ] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #2653 ? 
- [ ] tests added and passing

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -8,8 +8,8 @@ If you're interested in getting involved in the development of Modin, but aren't
 where start, take a look at the issues tagged `Good first issue`_ or Documentation_.
 These are issues that would be good for getting familiar with the codebase and better
 understanding some of the more complex components of the architecture. There is
-documentation here about the architecture_ that you will want to review in order to get
-started.
+documentation here about the :doc:`architecture </developer/architecture>` that you will
```
Comment (YarShev) [inline docs/contributing.rst:11]: I suppose this link should lead to `architecture.rst` opened but that doesn't happen. Please, check all the changed references.

**Review by YarShev — CHANGES_REQUESTED** [review]

File: docs/contributing.rst (hunk 1 above)
Comment (prutskov) [reply docs/contributing.rst:11]: No, this links work only as a separate site (https://modin.readthedocs.io/en/latest/). Sphinx dosn't provide browsing by internal links via github.

File: docs/contributing.rst [hunk 2]
```diff
@@ -250,11 +250,8 @@ More docs on this coming soon...
 
 .. _Good first issue: https://github.com/modin-project/modin/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue+%3Abeginner%3A%22
 .. _Documentation: https://github.com/modin-project/modin/issues?q=is%3Aissue+is%3Aopen+label%3A%22documentation+%3Abookmark_tabs%3A%22
-.. _architecture: architecture.html
-.. _internal methods:
 .. _black: https://github.com/ambv/black
 .. _flake8: http://flake8.pycqa.org/en/latest/
 .. _Github Actions: https://github.com/features/actions
-.. _testing:
```
Comment (YarShev) [inline docs/contributing.rst:258]: Why is this removed?

File: docs/developer/architecture.rst [hunk 3]
```diff
@@ -236,25 +236,21 @@ This is the list of execution frameworks and memory formats supported in Modin.
...
+- :doc:`Pyarrow on Ray </UsingPyarrowonRay/index>` (experimental)
     - Uses the Ray_ execution framework.
     - The compute kernel/in-memory format is a pyarrow Table.
 
-.. _pandas Dataframe: https://pandas.pydata.org/pandas-docs/version/0.23.4/generated/pandas.DataFrame.html
+.. _pandas Dataframe: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html
 .. _Arrow tables: https://arrow.apache.org/docs/python/generated/pyarrow.Table.html
 .. _Ray: https://github.com/ray-project/ray
 .. _code: https://github.com/modin-project/modin/blob/master/modin/engines/base/frame/data.py
-.. _Contributing: contributing.html
```
Comment (YarShev) [inline docs/developer/architecture.rst:253]: Please, fix `contributing` link above as for the rest (remove `..`).

File: docs/installation.rst [hunk 4]
```diff
@@ -84,7 +84,7 @@ You may already have a recent version of Dask_ installed, in which case you can
 Building Modin from Source
 --------------------------
 
-If you're planning on contributing_ to Modin, you will need to ensure that you are
+If you're planning on :doc:`contributing </contributing>` to Modin, you will need to ensure that you are
```
Comment (YarShev) [inline docs/installation.rst:87]: If a doc is located in the same folder, do we need to specify `/`?

File: docs/supported_apis/index.rst [hunk 5]
```diff
@@ -33,20 +33,17 @@ Modin.
...
 We have taken a community-driven approach to implementing new methods. We did a `study
 on pandas usage`_ to learn what the most-used APIs are. Modin currently supports **93%**
 of the pandas API based on our study of pandas usage, and we are actively expanding the
 API.
 
-.. _DataFrame: dataframe_supported.html
-.. _Series: series_supported.html
-.. _utilities: utilities_supported.html
-.. _I/O: io_supported.html
-.. _study on pandas usage: https://github.com/modin-project/study_kaggle_usage
```
Comment (YarShev) [inline docs/supported_apis/index.rst:50]: Why is this removed?

File: docs/contributing.rst (hunk 2 above)
Comment (prutskov) [reply docs/contributing.rst:258]: Because this link opens non-valid page for `testing` (this [page](https://github.com/airspeed-velocity/asv#airspeed-velocity), which are the next in links-list ).
image

File: docs/installation.rst (hunk 4 above)
Comment (prutskov) [reply docs/installation.rst:87]: Yes, in our case we need to specify all files by template `</path>` where `/ ` is `docs/` folder.

File: docs/developer/architecture.rst (hunk 3 above)
Comment (prutskov) [reply docs/developer/architecture.rst:253]: fixed

File: docs/supported_apis/index.rst (hunk 5 above)
Comment (prutskov) [reply docs/supported_apis/index.rst:50]: `DataFrame`, `Series`, `utilities` and `I/O` was changed using `:doc:` format. `study on pandas usage` is saved.

File: docs/contributing.rst [hunk 6]
```diff
@@ -214,6 +214,7 @@ Some details about using Modin on Ray with Asv:
 Some details for maintenance:
 
 * ``modin/asv_bench/asv.conf.json`` contains the modin dependencies, with which testing takes place.
+
```
Comment (YarShev) [inline docs/contributing.rst:217]: remove this extra line

File: docs/contributing.rst (hunk 2 above)
Comment (YarShev) [reply docs/contributing.rst:258]: Probably, we need to add a documentation page for testing because we don't have it. Can you open an issue for this?

File: docs/contributing.rst (hunk 6 above)
Comment (prutskov) [reply docs/contributing.rst:217]: This extra line is needed to fix sphinx warning message in building process.

File: docs/contributing.rst (hunk 2 above)
Comment (prutskov) [reply docs/contributing.rst:258]: Issue #2661  was created

Comment (YarShev) [reply docs/contributing.rst:258]: Thanks!

File: docs/contributing.rst (hunk 6 above)
Comment (YarShev) [reply docs/contributing.rst:217]: The sentence "We need to keep them up to date according to the dependencies in ``setup.py``." should be related to the sentence "``modin/asv_bench/asv.conf.json`` contains the modin dependencies, with which testing takes place.". Then you should rephrase these sentences to fit into one sentence.

Comment (YarShev) [reply docs/contributing.rst:217]: Okay, I see.

**Review by YarShev — APPROVED** [review]
@prutskov , LGTM, thanks!
